lib/llvmopencl/WorkItemAliasAnalysis.cc

```diff
@@ -116,10 +116,19 @@ WorkItemAAResult WorkItemAA::run(Function &F, AnalysisManager<Function> *AM) {
     return WorkItemAAResult(AM->getResult<WorkItemAA>(F));
 }
 
-bool WorkItemAliasAnalysis::runOnFunction(llvm::Function &) {
-    auto &TLIWP = getAnalysis<TargetLibraryInfoWrapperPass>();
-    Result.reset(new WorkItemAAResult(TLIWP.getTLI()));
-    return false;
+bool WorkItemAliasAnalysis::runOnFunction(
+#ifdef LLVM_OLDER_THAN_10_0
+    [[gnu::unused]]
```

Comment:
Please avoid using a GCC (GNU)-specific attribute here.

Reply:
* unsupported attrs are just ignored
* without this attr there will be a warning
* gnu attrs are supported by lots of compilers other than gcc, including clang and intel.

So IMHO we should keep it in order to reduce count of useless warnings.

Reply:
Fair enough! If it's supported by Clang too then it's better, but it'd be still cleaner to put it to CompilerWarnings.h as a macro which could be used for per-parameter warning suppression.

Reply:
If we are ready to require C++>=17, we can use standardized `[[maybe_unused]]`

Reply:
The minimum C++ standard version required by pocl is the one set by the oldest LLVM version. At least 7.0 can be still built with C++11 so that is the limit for the timing being.